arrows/core/metadata_map_io_csv.cxx

```diff
@@ -24,6 +31,49 @@ namespace arrows {
 
 namespace core {
 
+namespace {
+
+// ----------------------------------------------------------------------------
+template < typename out >
+void
+split( std::string const& s, char delim, out result )
```

Comment:
These feel like they either a) probably already exist, or at least b) ought to be in a utility module somewhere.